### PR TITLE
[Ripgrep] Add Completion Spec

### PR DESCRIPTION
**Pull request by Monkeyank**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature
**What is the current behavior? (You can also link to an open issue here)**
#103 
**What is the new behavior (if this is a feature change)?**
Adds basic Ripgrep completion spec
**Additional info:**

### CODE REVIEW

File: dev/rg.ts [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+      name: "--hidden",
+      description: "Searches hidden files and directories",
+    },
+    {
+      name: "--iglob",
+      description:
+        "Includes or excludes files and directories for searching that match given the glob",
+      args: {
+        name: "GLOB",
+        variadic: true,
```
Comment (QuiiBz) [inline dev/rg.ts:209]: Based on the description, I think this arg can have a `template: ["filepaths", "folders"]`

File: dev/rg.ts [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+        name: "Replacement Text",
+      },
+    },
+    {
+      name: ["-z", "--search-zip"],
+      description: "Search in compressed files",
+    },
+    {
+      name: ["-S, --smart-case"],
+      description: "Seaches case insensitively if the pattern is all lowercase",
```
Comment (QuiiBz) [inline dev/rg.ts:486]: Type at `Seaches` -> `Searches`

File: dev/rg.ts [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+            description:
+              "Sort by the last accessed time on a file. Always single-threaded",
+          },
+          {
+            name: "created",
+            description:
+              "Sort by the creation time on a file. Always single-threaded",
+          },
+        ],
+      },
```
Comment (QuiiBz) [inline dev/rg.ts:551]: I would love to see the args moved to a top-level const to avoid duplication!

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Excited about this spec! I left you some comments

File: dev/rg.ts [hunk 4]
```diff
@@ -0,0 +1,649 @@
...
+    },
+    {
+      name: "--colors",
+      description: "Specifies color settings for use in the output",
+      args: {
+        name: "Color Spec",
+        variadic: true,
+        description: "{type}:{attribute}:{value}, ex: `match:fg:magenta`",
```
[the remark this answers is not in the document]
Comment (Monkeyank) [reply dev/rg.ts:77]: The` {type}:{attribute}:{value}` isn't 3 separate arguments, rather it's one argument, and you can supply multiple of these.
Ex:  `rg --colors 'line:bg:yellow' 'match:fg:magenta'`

**Review by QuiiBz — APPROVED** [review]
LGTM!

**Pull request merged**